operator/pkg/name/name.go

```diff
@@ -110,6 +110,9 @@ const (
 	// Operator components
 	IstioOperatorComponentName      ComponentName = "IstioOperator"
 	IstioOperatorCustomResourceName ComponentName = "IstioOperatorCustomResource"
+
+	// Deprecated addon component name, used to exclude resources from pruning only.
+	AddonComponentName ComponentName = "AddonComponents"
```

Comment:
do we need to be explicit here? i.e. if a component name is unknown, we probably should not prune it anyway.

Reply:
ok good point